src/coreclr/nativeaot/BuildIntegration/Microsoft.NETCore.Native.targets

```diff
@@ -268,6 +268,9 @@ The .NET Foundation licenses this file to you under the MIT license.
       <IlcArg Condition="'$(_targetOS)' == 'win' and $(IlcMultiModule) != 'true' and '$(IlcGenerateWin32Resources)' != 'false' and '$(NativeLib)' != 'Static'" Include="--win32resourcemodule:%(ManagedBinary.Filename)" />
       <IlcArg Condition="$(IlcDumpIL) == 'true'" Include="--ildump:$(NativeIntermediateOutputPath)%(ManagedBinary.Filename).il" />
       <IlcArg Condition="$(NoWarn) != ''" Include='--nowarn:"$([MSBuild]::Escape($(NoWarn)).Replace(`%0A`, ``).Replace(`%0D`, ``))"' />
+      <IlcArg Condition="$(TreatWarningsAsErrors) == 'true'" Include="--warnaserror" />
+      <IlcArg Condition="$(WarningsAsErrors) != ''" Include='--enable-warnaserror:"$([MSBuild]::Escape($(WarningsAsErrors)).Replace(`%0A`, ``).Replace(`%0D`, ``))"' />
```

Comment:
Can we keep the style consistent? Most of these arguments are in the style of `nowarn`, `notrimwarn`, etc. Only a couple arguments that we share with crossgen2 have hyphens and spell out full words (crossgen2 is not consistent about the hyphen style either)

Can this be `--warnaserr` and `--nowarnaserr`? (I would prefer `warnaserror`/`nowarnaserror` but I fully expect that to be an uphill battle with System.CommandLine we don't need to fight.)

Reply:
I can definitely change the names, but as you mention, it would be hard to mimic what roslyn/illink does due to the command line parser in use.